aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -1242,16 +1242,21 @@ static void linalg_eigh_cusolver_syevj(const Tensor& eigenvalues, const Tensor&
   });
 }
 
-void linalg_eigh_cusolver(const Tensor& eigenvalues, const Tensor& eigenvectors, const Tensor& infos, bool upper, bool compute_eigenvectors) {
-  // TODO: syevj_batched should be added here, but at least for CUDA 11.2 it contains a bug leading to incorrect results
-  // See https://github.com/pytorch/pytorch/pull/53040#issuecomment-793626268 and https://github.com/cupy/cupy/issues/4847
+static void linalg_eigh_cusolver_syevj_batched(const Tensor& eigenvalues, const Tensor& eigenvectors, const Tensor& infos, bool upper, bool compute_eigenvectors) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(eigenvectors.scalar_type(), "linalg_eigh_cuda", [&] {
+    apply_syevj_batched<scalar_t>(eigenvalues, eigenvectors, infos, upper, compute_eigenvectors);
+  });
+}
 
+void linalg_eigh_cusolver(const Tensor& eigenvalues, const Tensor& eigenvectors, const Tensor& infos, bool upper, bool compute_eigenvectors) {
   // syevj is better than syevd for float32 dtype and matrix sizes 32x32 - 512x512
```

Comment:
this comment is misplaced (it refers to line 1256 now). Can you also add another comment with link to the data about why we are using syevjBatched for heuristics in line 1254?